exercises/binary-search/example.cpp

```diff
@@ -0,0 +1,23 @@
+#include "binary_search.h"
+
+int binary_search::find(const std::vector<int>& arr, int val) {
```

Comment:
shouldn't we return an iterator?

Reply:
I don't think we need to do like other languages. The idea is to learn the language, and the best practice is to return an iterator. I don't want to just use the std function because it teaches less.